rest/handler.go

```diff
@@ -266,24 +266,25 @@ func (h *handler) invoke(method handlerMethod, accessPermissions []Permission, r
 	// look up the database context:
 	var dbContext *db.DatabaseContext
 	if keyspaceDb != "" {
+		if shouldCheckAdminAuth {
+			// Confirm admin port is authorized before attempting to load db
+			authorized, err := h.checkAdminAuthenticationOnly()
```

Comment:
This looks like it's going to make two calls to the permissions endpoint for all admin requests (here and on line 426), which we want to avoid. (the previous usage of checkAdminAuthenticationOnly was only in the case where the database didn't exist and we were determining whether to return 404 or 403).